apps/launch/components/NominateContract/utils.tsx

```diff
@@ -0,0 +1,165 @@
+import { ReloadOutlined } from '@ant-design/icons';
+import { Alert, Button, Card, Flex, Spin, Typography } from 'antd';
+import Link from 'next/link';
+import React, { FC, ReactNode } from 'react';
+
+import { useAppSelector } from 'store/index';
+
+import { LogoSvg } from '../Layout/Logos';
+import { LoginV2 } from '../Login';
+import { GreenDot, LogoContainer, PurpleDot, RedDot, StyledMain } from './styles';
+
+const { Title, Text } = Typography;
+
+/************************ Presentational Components ************************/
```

Comment:
please do not put components in utils folders or files.
seperate concerns.
utils should be reserved for utility functions, helpers, .etc, things that are not tied to UI

Reply:
what do you think is better to name such files then, that contains component-helpers? especially when there are many ones ? maybe `components.tsx`? I don't think it's a good idea to keep everything close to the main component or create separate files

Reply:
> Where should smaller components be placed? I'm not a fan of creating separate files for each component as it causes too much clutter. Alternatively, we can keep them in the same file.

If they're small and truly unique to the main component, than leave them in the same file; preferably `index.tsx` if they're under a folder.

If they're large/complex (use your own determination here) and truly unique to the main component, then create another component in the same directory as the main component; i.e. 
`CrazyTable/index.tsx` (contains a main table, and some unique/small/non-usable components)
`CrazyTable/SuperBigFilter.tsx`

Otherwise, if they're not unique to the main component of the directory they're in, move them up the tree to their own component folder so it's clear they are reusable components. If we're making edits to the components, i.e. changing the width of a container, pass the values as props to the component.

> what do you think is better to name such file then, that contains component-helpers? especially when there are many ones ? maybe `components.tsx`? I don't think it's a good idea to keep everything close to the main component or create separate files

Helpers/Utils (non-ui functions, helper methods) that are specific to the component should be in the component file if they are specific to the component. If they are reusable, i.e. `someCommonMathFunction()` or `someCommonSortAlgorithm()`, put them in the root "utils" or "helpers" folder in a file named appropriately for the expected functions or helpers that live there. i.e. `utils/math/solana.ts` or `utils/math/evm.ts`

-----

further to this, the components folder should be for components that are _imported into pages_ as per the page's requirement. pages are where we should build the appropriate page. so we don't clutter the components folder with components that are clearly pages.

Reply:
please see how some larger orgs are organizing their projects, there is clear separation of concerns.

https://github.com/Uniswap/interface/tree/main/apps/web/src
https://github.com/DefiLlama/defillama-app/tree/main/src

Reply:
thanks for the links and explanation @truemiller, I'm more than agree about that the pages shouldn't be in the components, that helpers should be functions\hooks rather than jsx, I also have more other preferences, but that is how it is right now;
overall I think it is a tough and very controversial topic to decide in a PR, when we have many other apps where things are a little bit worse.

Can we raise this topic again at our upcoming FE eng meeting, where we can document our own decision?